control_algorithms/adrc.py

```diff
@@ -0,0 +1,62 @@
+"""
+Active Disturbance Rejection Control (ADRC) is a robust control strategy
+that estimates and compensates for disturbances in real-time without needing
+an explicit mathematical model of the system.
+
+It consists of:
+1. Tracking Differentiator (TD) - Smooths the reference signal
+2. Extended State Observer (ESO) - Estimates system states and disturbances
+3. Nonlinear State Error Feedback (NLSEF) - Generates the control signal
+
+Refer - https://en.wikipedia.org/wiki/Active_disturbance_rejection_control
+"""
+
+
+class ADRC:
+    def __init__(self, beta1: float, beta2: float, beta3: float, setpoint: float = 0.0):
+        """
+        Initialize the ADRC controller.
+
+        :param beta1: Gain for error correction in ESO
+        :param beta2: Gain for disturbance estimation in ESO
+        :param beta3: Gain for acceleration estimation in ESO
+        :param setpoint: Desired target value
+        """
+        self.beta1 = beta1
+        self.beta2 = beta2
+        self.beta3 = beta3
+        self.setpoint = setpoint
+
+        self.z1 = 0.0  # Estimated system output
+        self.z2 = 0.0  # Estimated system velocity
+        self.z3 = 0.0  # Estimated total disturbance
+
+    def compute(self, measured_value: float, dt: float) -> float:
```

Comment:
```suggestion
    def control_output(self, measured_value: float, dt: float) -> float:
```
or `get_control_output()` or `calculate_control_output()`